WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -106,12 +110,40 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     func cardPaymentTapped() {
         Task { @MainActor in
             showsCreatingOrderSheet = true
-            let order = try await createTestOrder()
+            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
             showsCreatingOrderSheet = false
-            let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+            try await collectPayment(for: order)
+        }
+    }
+
+    @MainActor
+    private func collectPayment(for order: Order) async throws {
+        let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+
+        // TODO: Here we should present something to show the payment was successful or not,
+        // and then clear the screen ready for the next transaction.
+    }
+
+    @MainActor
+    func totalsAppeared() async {
```

Comment:
nit: from the [doc of the `task` modifier](https://developer.apple.com/documentation/swiftui/view/task(priority:_:)), it seems like "will appear" matches more of the timing when this is called

```suggestion
    func totalsViewWillAppear() async {
```